include/nuttx/spinlock.h

```diff
@@ -32,6 +32,14 @@
 
 #include <nuttx/irq.h>
 
+#ifdef CONFIG_RW_SPINLOCK
+#include <stdatomic.h>
+typedef atomic_int rwlock_t;
```

Comment:
It can't.
There will be a error if we use normal type on Build / macOS (sim-02) 
```
(error: address argument to atomic operation must be a pointer to _Atomic type ('volatile rwlock_t *' (aka 'volatile int *') invalid))
```

So I used atomic type in the second commit of this PR but it seems not to work.

Reply:
Ok